### PR TITLE
docs: Manual performance test for logs

### PR DESCRIPTION
**Pull request by hisarbalik**
## Description

Changes proposed in this pull request (what was done and why):

- Instruction for test execution and prerequisites
- Preconfigured test scenarios and components
- Shell script deploy, execute, and fetch KPIs for each scenario
- Add Fluent-Bit version 2.2.1 and 2.2.2 test results
- Move initial metric performance test to the `docs/conributor/pocs` directory 

Changes refer to particular issues, PRs or documents:

- https://github.com/kyma-project/telemetry-manager/issues/765

## Traceability
- [x] The PR is linked to a GitHub issue.
- [ ] New features have a milestone set.
- [x] New features have defined acceptance criteria in a corresponding GitHub Issue, and all criteria are satisfied with this PR.
- [x] The corresponding GitHub issue has a respective `area` and `kind` label.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [x] Adjusted the documentation if the change is user-facing.
- [ ] The feature is unit-tested
- [ ] The feature is e2e-tested

### CODE REVIEW

File: docs/contributor/telemetry-load-test/README.md [hunk 1]
```diff
@@ -213,4 +213,75 @@ Each test scenario has its own test scripts responsible for preparing test scena
...
 
+</div>
+
+
+## Log Test (Fluent-Bit)
+
+### Assumptions
+
+The tests are executed for 20 minutes, so that each test case has a stabilized output and reliable KPIs.
+The Log test deploys a passive log producer ([Flog](https://github.com/mingrammer/flog)) and the log are collected by Fluent-Bit from each producer instance.
```
Comment (NHingerl) [inline docs/contributor/telemetry-load-test/README.md:224]:
```suggestion
The Log test deploys a passive log producer ([Flog](https://github.com/mingrammer/flog)), and the logs are collected by Fluent Bit from each producer instance.
```

File: docs/contributor/telemetry-load-test/README.md [hunk 2]
```diff
@@ -213,4 +213,75 @@ Each test scenario has its own test scripts responsible for preparing test scena
...
+The test setup deploys 40 individual log producer Pods; each which produces ~100 MByte logs.
```
Comment (NHingerl) [inline docs/contributor/telemetry-load-test/README.md:225]:
```suggestion
The test setup deploys 40 individual log producer Pods; each of which produces ~100 MByte logs. 
```

File: docs/contributor/telemetry-load-test/README.md [hunk 3]
```diff
@@ -213,4 +213,75 @@ Each test scenario has its own test scripts responsible for preparing test scena
...
+The test setup deploys 40 individual log producer Pods; each which produces ~100 MByte logs. 
+
+The following test cases are identified:
+
+1. Test average throughput end-to-end.
+2. Test buffering and retry capabilities of LogPipeline with simulated backend outages.
+3. Test average throughput with 3 LogPipelines simultaneously end-to-end.
+4. Test buffering and retry capabilities of 3 LogPipeline with simulated backend outages.
+
+Backend outages simulated with Istio Fault Injection, 70% of traffic to the Test Backend will return `HTTP 503` to simulate service outages.
```
Comment (NHingerl) [inline docs/contributor/telemetry-load-test/README.md:234]:
```suggestion
Backend outages are simulated with Istio Fault Injection, 70% of traffic to the test backend will return `HTTP 503` to simulate service outages.
```

File: docs/contributor/telemetry-load-test/README.md [hunk 4]
```diff
@@ -213,4 +213,75 @@ Each test scenario has its own test scripts responsible for preparing test scena
...
+
+### Setup
+
+The following diagram shows the test setup used for all test cases.
+
+![LogPipeline Test Setup](./assets/log_perf_test_setup.drawio.svg)
+
+In all test scenarios, a preconfigured trace load generator is deployed on the test cluster.
+
+A Prometheus instance is deployed on the test cluster to collect relevant metrics from Fluent-Bit instances and to fetch the metrics at the end of the test as test scenario result.
```
Comment (NHingerl) [inline docs/contributor/telemetry-load-test/README.md:244]:
```suggestion
A Prometheus instance is deployed on the test cluster to collect relevant metrics from Fluent Bit instances and to fetch the metrics at the end of the test as test scenario result.
```

File: docs/contributor/telemetry-load-test/README.md [hunk 5]
```diff
@@ -213,4 +213,75 @@ Each test scenario has its own test scripts responsible for preparing test scena
...
+
+All test scenarios also have a test backend deployed to simulate end-to-end behaviour.
+
+Each test scenario has its own test scripts responsible for preparing test scenario and deploying on test cluster, running the scenario, and fetching relevant metrics/KPIs at the end of the test run. After the test, the test results are printed out.
```
Comment (NHingerl) [inline docs/contributor/telemetry-load-test/README.md:248]:
```suggestion
Each test scenario has its own test scripts responsible for preparing the test scenario and deploying it on the test cluster, running the scenario, and fetching relevant metrics and KPIs at the end of the test run. After the test, the test results are printed out.
```

File: docs/contributor/telemetry-load-test/assets/log-fluentbit-max-pipeline.yaml [hunk 6]
```diff
@@ -0,0 +1,34 @@
+apiVersion: telemetry.kyma-project.io/v1alpha1
+kind: LogPipeline
+metadata:
+  name: load-test-2
+spec:
+  output:
+    http:
+      dedot: false
```
Comment (chrkl) [inline docs/contributor/telemetry-load-test/assets/log-fluentbit-max-pipeline.yaml:8]: Let's enable de-dotting. Even though, it's not required by the backend running a Lua script is something with a potential performance impact and where we might have regressions.

File: docs/contributor/telemetry-load-test/README.md [hunk 7]
```diff
@@ -213,4 +213,75 @@ Each test scenario has its own test scripts responsible for preparing test scena
...
+|        2.2.1 |                  7294                   |                   3847                   |              80553              |       185,181        |    0.8,0.9    |                  2484                   |                   1205                   |              94435              |       321,325        |    0.8,0.8    |                  7611                   |                   665                    |              36704              |       208,219        |    0.8,0.8    |                  2501                   |                   231                    |              97754              |       325,320        |    0.8,0.8    |
```
Comment (chrkl) [inline docs/contributor/telemetry-load-test/README.md:284]: We see a output rate that is lower than the input rate and a significant file-system buffer usage already for the first test. It looks like this test case already creates a backpressure situation. Maybe try to reduce the generated amount of logs a bit.

File: docs/contributor/telemetry-load-test/README.md [hunk 8]
```diff
@@ -16,7 +16,7 @@ This document describes a reproducible test setup to determine the limits and KP
 
 All test scenarios use a single test script [run-load-test.sh](assets/run-load-test.sh), which provides following parameters: 
 
-- `-t` The test target type supported values are `traces, metrics, metricagent`, default is `traces`
+- `-t` The test target type supported values are `traces, metrics, metricagent, fluentbit`, default is `traces`
```
[the remark this answers is not in the document]
Comment (hisarbalik) [reply docs/contributor/telemetry-load-test/README.md]: target renamed

File: docs/contributor/telemetry-load-test/assets/log-fluentbit-max-pipeline.yaml (hunk 6 above)
Comment (hisarbalik) [reply docs/contributor/telemetry-load-test/assets/log-fluentbit-max-pipeline.yaml]: dedot enabled

File: docs/contributor/telemetry-load-test/README.md (hunk 7 above)
Comment (hisarbalik) [reply docs/contributor/telemetry-load-test/README.md]: test result updated with new setup

**Review by NHingerl — APPROVED** [review]

**Review by chrkl — APPROVED** [review]

**Pull request merged**